pkg/provider/k8s/k8s.go

```diff
@@ -31,23 +31,140 @@ type K8s struct {
 	ctx context.Context
 }
 
+// K8sClient holds the fields used to generate API reqeust from within a cluster.
+type K8sClient struct {
+	// The k8s provider used when we work with the manifest files.
+	k8sProvider *K8s
+	// DeploymentFiles files provided from the cli.
+	DeploymentFiles []string
+	// Vaiables to subtitude in the DeploymentFiles.
+	// These are also used when the command requires some variables that are not provided by the deployment file.
+	DeploymentVars map[string]string
+	// DeploymentFile content after substituting the variables filename is used as the map key.
+	deploymentsContent []provider.ResourceFile
+
+	ctx context.Context
+}
+
 // New returns a k8s client that can apply and delete resources.
-func New(ctx context.Context, config clientcmdapi.Config) (*K8s, error) {
-	restConfig, err := clientcmd.NewDefaultClientConfig(config, &clientcmd.ConfigOverrides{}).ClientConfig()
+func New(ctx context.Context, clientset *kubernetes.Clientset) *K8s {
+	return &K8s{
+		ctx: ctx,
+		clt: clientset,
+	}
+}
+
+// NewK8sClient returns a k8s client that can apply and delete resources.
+func NewK8sClient() (*K8sClient, error) {
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		return nil, errors.Wrapf(err, "config error")
+		return nil, errors.Wrapf(err, "k8s config error")
 	}
-	clientset, err := kubernetes.NewForConfig(restConfig)
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		return nil, errors.Wrapf(err, "client error")
+		return nil, errors.Wrapf(err, "k8s client error")
 	}
 
-	return &K8s{
-		ctx: ctx,
-		clt: clientset,
+	ctx := context.Background()
+	return &K8sClient{
+		k8sProvider:    New(ctx, clientset),
+		DeploymentVars: make(map[string]string),
+		ctx:            ctx,
 	}, nil
 }
 
+// DeploymentsParse parses the deployment files and saves the result as bytes grouped by the filename.
+func (c *K8sClient) DeploymentsParse(*kingpin.ParseContext) error {
+
+	deploymentsContent, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		log.Fatalf("Couldn't parse deployment files: %v", err)
+	}
+
+	c.deploymentsContent = deploymentsContent
+	return nil
+}
+
+// Scale scales a deployment to a given number of replicas.
+func (c *K8sClient) Scale(replicas *int32) error {
+
+	for _, deployment := range c.deploymentsContent {
+		separator := "---"
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+
+		for _, text := range strings.Split(string(deployment.Content), separator) {
+			text = strings.TrimSpace(text)
+			if len(text) == 0 {
+				continue
+			}
+
+			resource, _, err := decode([]byte(text), nil, nil)
+			if err != nil {
+				return errors.Wrapf(err, "decoding the resource file:%v, section:%v...", deployment.Name, text[:100])
+			}
+			if resource == nil {
+				continue
+			}
+
+			switch kind := strings.ToLower(resource.GetObjectKind().GroupVersionKind().Kind); kind {
+			case "deployment":
```

Comment:
I would use an `if` statement here. for me a switch indicates that we might handle more cases in the future which is not the case with the scaling which can be done only for deployments.